emcc.py

```diff
@@ -3072,7 +3072,9 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
   if shared.Settings.USE_PTHREADS and shared.Settings.ALLOW_MEMORY_GROWTH:
     final = shared.Building.apply_wasm_memory_growth(final)
 
-  if options.opt_level >= 2 and options.debug_level <= 2:
+  # Do not do inter wasm and js minification for dynamic linking because it prevents us from using the
+  # exported wasm functions from Module["asm"] directly which gives faster startup and runtime performance
+  if options.opt_level >= 2 and options.debug_level <= 2 and not shared.Settings.RELOCATABLE:
```

Comment:
There is already a bunch of logic inside `minify_wasm_js` for avoiding certain optimizations.   I think we should extent that logic rather than add an additional condition here. 

For example it looks like we already skip the minification if `Settings.LINKABLE` is set.

Reply:
Ok, I moved the check to `minify_wasm_js`.